webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/services/nodes/NodeInfoService.scala

```diff
@@ -552,9 +555,11 @@ trait NodeInfoServiceCached extends NodeInfoService with NamedZioLogger with Cac
     }
   }.toBox
 
-  def getNodeInfo(nodeId: NodeId): Box[Option[NodeInfo]] = withUpToDateCache(s"${nodeId.value} node info") { cache =>
+  def getNodeInfo(nodeId: NodeId): Box[Option[NodeInfo]] = getNodeInfoPure(nodeId).toBox
+
+  def getNodeInfoPure(nodeId: NodeId): IOResult[Option[NodeInfo]] = withUpToDateCache(s"${nodeId.value} node info") { cache =>
```

Comment:
Why 'Pure' ?? I know you want to distinguish from the 'Box' function and cannot use the same name  but it's could lead to think that this is a pure function (and it's not)

Reply:
It's pure, it's just a value. Effects are not done until `.runNow` is called

Reply:
I see your point, it wil always give an IO Result, and do nothing.